tests/core/full_node/stores/test_coin_store.py

```diff
@@ -230,6 +244,179 @@ async def test_num_unspent(self, bt, db_version):
 
         assert test_excercised
 
+    # Utility method to verify that the number of coins matches the expected number
+    # This method tests chia.wallet.wallet_coin_store.WalletCoinStoreWallet.get_all_unspent_coins()
+    async def verify_coin_count(self, current_wsm, expected_unspent_count):
+        all_unspent_coins: Set[WalletCoinRecord] = await current_wsm.coin_store.get_all_unspent_coins()
+
+        # Make sure each coin is actually unspent
+        for coin in all_unspent_coins:
+            assert coin.spent == False
+            assert coin.spent_block_height == 0 
+
+        # Count the number of unspent coins, skipping zero-value coins
+        actual_unspent_count = 0
+        for coin in all_unspent_coins:
+            if coin.spent == False and coin.coin.amount != 0:
+                actual_unspent_count += 1
+
+        # Verify the correct number of unspent coins
+        assert actual_unspent_count == expected_unspent_count
+
+        # As a sanity check, obtain all coins from the coin store
+        all_coins: Set[WalletCoinRecord] = await current_wsm.coin_store.get_all_coins()
+        
+        # Calculate the number of unspent coins obtained from get_all_coins
+        # and compare with the previous number. Again skip zero-value coins.
+        all_coins_actual_unspent_count = 0
+        for coin in all_coins:
+            if coin.spent == False and coin.coin.amount != 0:
+                all_coins_actual_unspent_count += 1
+
+        # Sanity check to compare with above
+        assert all_coins_actual_unspent_count == actual_unspent_count == expected_unspent_count
+
+    # This will test that chia.wallet.wallet_coin_store.WalletCoinStoreWallet.get_all_unspent_coins()
+    # is working as designed. It uses two wallets: a farm wallet and a receive wallet.
+    #
+    # Part 1: Farm a block with the farm wallet.
+    #         Farm wallet has 2 coins, receive wallet has 0.
+    # Part 2: Send 1 mojo from farm wallet to receive wallet and farm another block.
+    #         Farm wallet has 6 coins (5 coinbase + 1 change), receive wallet has 1.
+    # Part 3: Send all money from farm wallet to receive wallet in two coins and farm another block.
+    #         Farm wallet has 2 coins, receive wallet has 3.
+    # Part 4: Send all money from receive wallet to farm wallet and farm another block.
+    #         Farm wallet has 5 coins (4 coinbase + 1 received), receive wallet has 0.
+    @pytest.mark.asyncio
+    async def test_get_all_unspent_coins(
```

Comment:
this test might be better if it didn't have wallets and nodes. Check  out the other tests in the file. We can usually just add coins manually to the coin store, and don't need all these services up and running, which takes time and decreases stability

Reply:
Added new issue https://github.com/Chia-Network/chia-blockchain/issues/13244